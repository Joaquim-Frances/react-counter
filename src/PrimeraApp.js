import { Fragment } from "react";
import PropTypes from 'prop-types';


const PrimeraApp = ( { saludo, subtitulo }) => {

    return (
        <Fragment>
            <h1> { saludo } </h1>
            <p> { subtitulo } </p>
        </Fragment>
    )

}

PrimeraApp.propTypes = {

    saludo: PropTypes.string.isRequired,

}

PrimeraApp.defaultProps = {
    saludo: 'Hola World, is getting dark downhere...',
    subtitulo: 'Hola soy un subtitulo peque√±in.',
}

export default PrimeraApp;