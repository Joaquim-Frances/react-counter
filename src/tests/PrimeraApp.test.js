import '@testing-library/jest-dom';
import PrimeraApp from "../PrimeraApp";
import React from 'react';
import { shallow } from 'Enzyme'


describe('Pruebas en PrimeraApp', () => {
    

    test('debe mostrar <PrimeraApp /> correctamente', () => {
        const saludo='Hola caracola';
        const wrapper = shallow( <PrimeraApp saludo={ saludo }/>);

        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar el subtítulo enviado por props', () => {
        
        const saludo='Hola caracola';
        const subtitulo = 'Hey soy el subtítulo más molón';
        const wrapper = shallow( 
            <PrimeraApp 
                saludo={ saludo }
                subtitulo ={ subtitulo }
            />
        );

        const textParrafo = wrapper.find('p').text();
        expect(textParrafo).toBe(subtitulo);
    })
    
})
