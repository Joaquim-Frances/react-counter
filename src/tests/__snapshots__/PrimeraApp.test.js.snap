// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp debe mostrar <PrimeraApp /> correctamente 1`] = `
<div>
  <h1>
     
    Hola caracola
     
  </h1>
  <p>
    Hola soy un subtitulo peque√±in.
  </p>
</div>
`;
